state/resources_state_resource.go

```diff
@@ -312,7 +331,16 @@ func (st resourceState) OpenResource(applicationID, name string) (resource.Resou
 		return resource.Resource{}, nil, errors.NotFoundf("resource %q", name)
 	}
 
-	resourceReader, resSize, err := st.storage.Get(storagePath)
+	var resourceReader io.ReadCloser
+	var resSize int64
+	switch resourceInfo.Type {
+	case charmresource.TypeDocker:
```

Comment:
Oh, regarding my comment about the reader in state layer - I see that's an existing way we do it, so best to be consistent. So ignore that other comment.